src/components/Signup/Signup.jsx

```diff
@@ -0,0 +1,255 @@
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import Joi from 'joi-browser';
+import PropTypes from 'prop-types';
+import signupInputSchema from '../../schemas/signup';
+import signupAction from '../../actions/user/signupAction';
+import Form from '../Form/Form';
+import Input from '../Input/Input';
+import Button from '../Button/Button';
+import GoogleIcon from '../../../assets/icon/google.png';
+import TwitterIcon from '../../../assets/icon/twitter.png';
+import FacebookIcon from '../../../assets/icon/facebook.png';
+import './Signup.scss';
+
+/**
+ *
+ *
+ * @export
+ * @class Signup
+ * @extends {Component}
+ */
+export class Signup extends Component {
+  state = {
+    data: {
+      firstName: '',
+      lastName: '',
+      userName: '',
+      email: '',
+      password: '',
+      confirmPassword: '',
+    },
+    errors: {},
+  };
+
+  getValidationErrorMessages = (error) => {
+    const validationErrors = {};
+    const errorDetails = error.details;
+
+    for (let i = errorDetails.length - 1; i >= 0; i -= 1) {
+      const errorDetail = errorDetails[i];
+      const input = errorDetail.path[0];
+      validationErrors[input] = errorDetail.message;
+    }
+
+    return validationErrors;
+  };
+
+  validateInputs = () => {
+    const { data } = this.state;
+    const VALIDATION_OPTION = { abortEarly: false };
+
+    const { error } = Joi.validate(data, signupInputSchema, VALIDATION_OPTION);
+
+    if (!error) return null;
+    return this.getValidationErrorMessages(error);
+  };
+
+  renderSignupError = () => {
+    const { errors } = this.state;
+    const {
+      signupReducer: { error },
+    } = this.props;
+
+    errors.signup = error;
+    this.setState({ errors });
+  };
+
+  redirectUserToProfilePage = () => {
+    const {
+      history: { push },
+    } = this.props;
+    push('/profile');
+  };
+
+  handleChange = ({ target }) => {
+    const { data } = this.state;
+    data[target.name] = target.value;
+    this.setState({ data });
+  };
+
+  clearErrorMessages = ({ target }) => {
+    const { errors } = this.state;
+    errors[target.name] = '';
+    this.setState({ errors });
+  };
+
+  handleSubmit = async (event) => {
+    event.preventDefault();
+    const { data: user } = this.state;
+    const { signupAction: signup } = this.props;
+
+    const validationErrors = this.validateInputs();
+    this.setState({
+      errors: validationErrors || {},
+    });
+
+    if (validationErrors) return null;
+
+    await signup(user);
+
+    const { signupReducer } = this.props;
+    if (signupReducer.error) this.renderSignupError();
+    if (signupReducer.signedUp) this.redirectUserToProfilePage();
+  };
+
+  /**
+   *
+   *
+   * @returns
+   * @memberof Signup
+   */
+  render() {
+    const { data, errors } = this.state;
+    const { signupReducer } = this.props;
+
+    return (
+      <>
+        <main className="Signup">
+          <section className="Signup__info">
```

Comment:
There's a possibility we might need to abstract this into it's own component, as I've noticed that many other pages have this